### PR TITLE
Issue 239 2

### PR DESCRIPTION
**Pull request by drumwhizz**
Added the ability to return low level errors from serial port read/writes to the application layer, in order to handle USB disconnection events.

### CODE REVIEW

**Review by RobHumphris — APPROVED** [review]

File: ublox-bluetooth.go [hunk 1]
```diff
@@ -249,6 +250,12 @@ func (ub *UbloxBluetooth) serialportReader() {
 
 	for {
 		select {
+		case err := <-echan:
+			go func() {
+				// Check for catastophic failure
+				ub.errorHandler(err)
+			}()
+			return
```
Comment (drumwhizz) [inline ublox-bluetooth.go:255]: Strictly this should not be an a go function, it should be the responsibility of the error handler function to decide whether the error handling warrants a separate go function.

File: ublox-bluetooth.go [hunk 2]
```diff
@@ -317,7 +324,15 @@ func (ub *UbloxBluetooth) Write(data string) error {
...
 		b = []byte(append([]byte(data), tail...))
 	}
-	return ub.WriteBytes(b)
+	err := ub.WriteBytes(b)
+
+	if err != nil {
+		go func() {
+			// Check for catastophic failure
+			ub.errorHandler(err)
+		}()
```
Comment (drumwhizz) [inline ublox-bluetooth.go]: Strictly this should not be an a go function, it should be the responsibility of the error handler function to decide whether the error handling warrants a separate go function.

File: ublox-bluetooth.go (hunk 1 above)
Comment (drumwhizz) [reply ublox-bluetooth.go:255]: done

File: ublox-bluetooth.go (hunk 2 above)
Comment (drumwhizz) [reply ublox-bluetooth.go]: done

**Pull request merged**